### ISSUE TITLE
Locale set through slug not persisted in cookie/session

### ISSUE THREAD

**Issue by oscaralexander** [codezero-be/laravel-localized-routes#48]
First of all, thanks for this beautiful package, Ivan. It requires some configuration, but the helpers feel very intuitive; almost like they should be native.

I'm running into something unexpected, however. Please consider the following setup:

```
// config/localized-routes.php

return [
    'supported-locales' => ['en', 'nl'],
    'omit_url_prefix_for_locale' => null,
    'redirect_to_localized_urls' => true,
    'redirect_status_code' => 301,
    '404_view' => 'errors.404',
    'use_locale_middleware' => true,
    'use_localizer' => false, 
];

// routes.web.php

Route::localized(function () {
    Route::get('/', function () {
        return view('welcome');
    });
});

Route::fallback(\CodeZero\LocalizedRoutes\Controller\FallbackController::class)
    ->middleware(\CodeZero\LocalizedRoutes\Middleware\SetLocale::class);
```

Whenever I visit `/` I get redirected to `/nl`. This in itself is surprising given that `use_localizer` is set to `false` and the configured default app locale is `en`. What's stranger still is that whenever I request `/en` (which I believe should set and persist the `en` locale), and subsequently request `/`, it does not redirect me to `/en` but to `/nl` again. (The same happens when `use_localizer` is enabled.)

In other words, the locale (app, cookie, and session) does not seem to get updated by visiting a slugged URL. Perhaps, I'm missing something, but I feel I've RTFM'ed a few times now 😅

**Comment by ivanvermeyen** [comment]
Hello,

That is indeed strange!
Does the same thing happen with all routes, like "/about" redirecting to "/nl/about"?
Or is it just the home route?

**Comment by oscaralexander** [comment]
Just tried it with an `/about` route as you suggested.

```
Route::get('about', function () {
    return view('about');
});
```
Redirects happen as follows, in this order:

1. `/about` → `/en/about` (desirable/expected)
2. `/nl/about` → `/nl/about` (desirable/expected)
3. `/about` → `/en/about` (not desirable/expected)

Oddly, while `/about` redirects to `/en/about`, the homepage `/` keeps redirecting redirects to `/nl/`. With `use_localizer` enabled, the `locale` cookie and session get set, but somehow forgotten or overridden by a default on a subsequent 'slugless' request.

**Comment by ivanvermeyen** [comment]
Did you add the middleware to the `$middlewarePriority` array in the `Http/Kernel.php`?
Perhaps the middleware now runs before the session get activated or something...

**Comment by oscaralexander** [comment]
I was wondering about that... My `Http/Kernel.php` (Laravel 8.65) has no such array, only `$middleware`, `$middlewareGroups`, and `$routeMiddleware` 🤔

**Comment by ivanvermeyen** [comment]
Ah yes.
They moved it to the parent class, but in some cases you still need it.
When you want to use route model binding you will definitely need it too.
You can just copy it over and add the middleware between the sessions and bindings middleware.

**Comment by oscaralexander** [comment]
Done! The problem persists, unfortunately.

Mind you, this is a clean Laravel installation. Getting localized routes was literally the first thing I did, so there's no conflicting middleware or other stuff going on.

**Comment by ivanvermeyen** [comment]
Oh ok, can you put it somewhere so i can have look?

**Comment by oscaralexander** [comment]
Sure Ivan, and thanks for your time and effort. You can find the brand new repo here :)

https://gitlab.com/ufomeldpunt/ufodb

**Comment by ivanvermeyen** [comment]
I don't know what changed, but weird things are happening.
I created a fresh app myself too.

If you go to `/en` or `/nl` the locale is stored in the session and in a cookie.

The FIRST time you trigger a redirect, it takes the correct locale from the session/cookie.
But after that, even tho the session and cookie changes, the redirect is cached or remembered somehow...
Even if I comment out the redirect code, or do a `dd()` it has no effect at all... It is still redirecting...

Also, each redirect is potentially cached for a different locale:

- Go to `/nl` - sets nl locale
- Go to `/` - redirects to `/nl`
- 
- Go to `/en` - sets en locale
- Go to `/about` - redirects to `/en/about`
- 
- Go to `/` - still redirects to `/nl`

Is Laravel caching the redirect response?
I'll have to investigate deeper.

**Comment by oscaralexander** [comment]
> Is Laravel caching the redirect response?

Seems so! I'll keep an eye on this issue. Hope you find some time to look into it. Thanks and stay safe!

**Comment by ivanvermeyen** [comment]
I think adding no-cache headers to the redirect fixed it.
Maybe this problem was here all along, but no one caught it yet... I don't know. :)
Give it a try in 2.6.2.

**Comment by oscaralexander** [comment]
That solved it! I had to [clear Chrome's the permanent redirect cache](https://stackoverflow.com/a/35093587), but once that was done everything worked as expected. Thanks so much!

**Comment by ivanvermeyen** [comment]
Great!
That was a brain breaker, yet such an easy fix :)
